src/offset/local/unix.rs

```diff
@@ -68,19 +68,18 @@ struct Cache {
     last_checked: SystemTime,
 }
 
-#[cfg(target_os = "android")]
-const TZDB_LOCATION: &str = " /system/usr/share/zoneinfo";
-
 #[cfg(target_os = "aix")]
 const TZDB_LOCATION: &str = "/usr/share/lib/zoneinfo";
 
-#[allow(dead_code)] // keeps the cfg simpler
 #[cfg(not(any(target_os = "android", target_os = "aix")))]
 const TZDB_LOCATION: &str = "/usr/share/zoneinfo";
 
 fn fallback_timezone() -> Option<TimeZone> {
     let tz_name = iana_time_zone::get_timezone().ok()?;
+    #[cfg(not(target_os = "android"))]
     let bytes = fs::read(format!("{}/{}", TZDB_LOCATION, tz_name)).ok()?;
+    #[cfg(target_os = "android")]
+    let bytes = android_tzdata::find_tz_data(&tz_name).ok()?;
```

Comment:
In PR code for `src/offset/local/tz_info/timezone.rs` (above) the `fn from_posix_tz` code attempts do things the "Android Way" but if that doesn't succeed it falls back to the current implementation. Specifically, if `android_tzdata::find_tz_data(tz_string)` fails it will fall through and use the current general implementation.

Should `fn fallback_timezone` code allow a similar fallback? e.g.

```lang-rust
fn fallback_timezone() -> Option<TimeZone> {
    let tz_name = iana_time_zone::get_timezone().ok()?;
    #[cfg(target_os = "android")]
    if let Ok(bytes) = android_tzdata::find_tz_data(&tz_name).ok() {
        return TimeZone::from_tz_data(&bytes).ok();
    }
    let bytes = fs::read(format!("{}/{}", TZDB_LOCATION, tz_name)).ok()?;
    TimeZone::from_tz_data(&bytes).ok()
}
```

Reply:
In `from_posix_tz()`, we try to parse the tz string as a transition rule as a fallback, which should or shouldn't work regardless of the OS. On the other hand, trying to find a tz file on Android won't ever work, as far as I know, so I don't think it makes sense as a fallback.

Reply:
Okay. 👍